overlord/snapstate/snapmgr.go

```diff
@@ -21,38 +21,100 @@
 package snapstate
 
 import (
+	"fmt"
+
+	"github.com/ubuntu-core/snappy/i18n"
 	"github.com/ubuntu-core/snappy/overlord/state"
+	"github.com/ubuntu-core/snappy/progress"
+	"github.com/ubuntu-core/snappy/snappy"
 )
 
+// Install initiates a change installing snap.
+// Note that the state must be locked by the caller.
+func Install(change *state.Change, snap, channel string) error {
+	t := change.NewTask("install-snap", fmt.Sprintf(i18n.G("Installing %q"), snap))
+	t.Set("name", snap)
+	t.Set("channel", channel)
+
+	return nil
+}
+
+// Remove initiates a change removing snap.
+// Note that the state must be locked by the caller.
+func Remove(change *state.Change, snap string) error {
+	t := change.NewTask("remove-snap", fmt.Sprintf(i18n.G("Removing %q"), snap))
+	t.Set("name", snap)
+
+	return nil
+}
+
+type backendIF interface {
+	Install(name, channel string, flags snappy.InstallFlags, meter progress.Meter) (string, error)
+	Remove(name string, flags snappy.RemoveFlags, meter progress.Meter) error
+}
+
+type snappyBackend struct {
```

Comment:
Can we rename this to `defaultBackend` and the interface to `managerBackend`?

Also, let's please use the standard `type foo struct{}` style for empty structs.